src/package/packageVersion.ts

```diff
@@ -336,6 +343,47 @@ export class PackageVersion {
     }
   }
 
+  /**
+   * Query the Package2Version SObject and return data with the provided type.
+   *
+   * NOTE: There is a limit of 2000 records that can be returned, otherwise
+   * a GACK might be thrown. If more than 2000 records are desired you should
+   * filter the query by date and aggregate all results.
+   *
+   * @param connection jsForce Connection to the org.
+   * @param fields The fields to include in the returned data. Defaults to all fields.
+   * @param whereClause Optional where clause to filter the query.
+   * @param whereClauseItems Optional array of where clause items to match. The query is chunked,
+   * meaning broken into multiple queries when the query length would exceed the maximum char limit.
+   * When defining items here, the `whereClause` argument must use this token for the item
+   * replacement: `'%IDS%'`.
+   * @param orderBy Optional order by clause. Defaults to LastModifiedDate descending.
+   * @returns Results from querying the Package2Version SObject.
+   */
+  public static async queryPackage2Version<T extends Record<string, unknown> = Record<string, unknown>>(
```

Comment:
why is this generic?  isn't the type for Package2Version concrete?

Reply:
also, if this is a brand new API, I think destructuring the params to an typed object is nicer since there are a bunch of optional ones

Reply:
It's generic because the return shape doesn't necessarily match `Package2Version` and could be a subset of it.  I really just wanted a way for the caller to define the type and not just be `Partial<Package2Version>`.  There's probably a better way though.

As for the params, it did grow from what I originally expected so an options object makes sense.  I can change that.